src/core/api/getSwapLiteQuote.ts

```diff
@@ -0,0 +1,71 @@
+import type { SwapError, SwapUnit } from '../../swap/types';
+import { isSwapError } from '../../swap/utils/isSwapError';
+import type { Token } from '../../token';
+import { formatTokenAmount } from '../utils/formatTokenAmount';
+import { getSwapQuote } from './getSwapQuote';
+import type {
+  APIError,
+  GetSwapQuoteParams,
+  GetSwapQuoteResponse,
+} from './types';
+
+type GetSwapLiteQuoteResponse = {
+  response?: GetSwapQuoteResponse;
+  error?: APIError;
+  formattedFromAmount?: string;
+};
+
+type GetSwapLiteQuoteParams = Omit<GetSwapQuoteParams, 'from'> & {
+  fromSwapUnit: SwapUnit;
+  from?: Token;
+};
+
+/**
+ * Fetches a quote for a swap, but only if the from and to tokens are different.
+ */
+
+export async function getSwapLiteQuote({
+  amount,
+  amountReference,
+  from,
+  maxSlippage,
+  to,
+  useAggregator,
+  fromSwapUnit,
+}: GetSwapLiteQuoteParams): Promise<GetSwapLiteQuoteResponse> {
+  // only fetch quote if the from token is provided
+  if (!from) {
+    return { response: undefined, formattedFromAmount: '', error: undefined };
+  }
+
+  let response: GetSwapQuoteResponse | undefined;
+  // only fetch quote if the from and to tokens are different
+  if (to?.symbol !== from?.symbol) {
+    response = await getSwapQuote({
+      amount,
+      amountReference,
+      from,
+      maxSlippage,
+      to,
+      useAggregator,
+    });
+  }
+
+  let formattedFromAmount = '';
+  if (response && !isSwapError(response)) {
+    formattedFromAmount = response?.fromAmount
+      ? formatTokenAmount(response.fromAmount, response.from.decimals)
+      : '';
+
+    fromSwapUnit.setAmountUSD(response?.fromAmountUSD || '');
+    fromSwapUnit.setAmount(formattedFromAmount || '');
```

Comment:
if this is going to be a core/api, it shouldn't setState, is there a reason a consumer would use this vs. getSwapQuote?

Reply:
oh yeah good catch let me move it to `swap`. no reason to have this here.